src/extension.ts

```diff
@@ -519,30 +519,39 @@ class Device extends vscode.TreeItem {
     }
     
     async captureScreenshot() {
-        const error = (message) => {
-            vscode.window.showErrorMessage("Error capturing screenshot: " + message);
+        let anyErrors = false;
+        const handleCaptureError = (e) => {
+            anyErrors = true;
+            vscode.window.showErrorMessage("Error capturing screenshot: " + (e.message || e));
         }
-        
+
         try {
             const screenshotDirectory = await getSharedTempDir('ev3dev-screenshots');
             const screenshotFile = `${screenshotDirectory}/ev3dev-${sanitizedDateString()}.png`;
 
             const conn = await this.exec('fbgrab -');
             const writeStream = fs.createWriteStream(screenshotFile);
+
+            conn.on('error', (e: Error) => handleCaptureError);
+
             writeStream.on('open', () => {
                 conn.stdout.pipe(writeStream);
             });
 
             writeStream.on('close', () => {
-                vscode.window.setStatusBarMessage(`Screenshot successfully captured`);
-                vscode.commands.executeCommand('vscode.open', vscode.Uri.file(screenshotFile));
+                if(!anyErrors) {

```

Comment:
> can we use 'finish' instead of 'close' for success *so we don't have to have the anyErrors variable?*

(emphasis mine)

If the `finish` event fires after an error is thrown, it doesn't meet this criterion.